### PR TITLE
Storydocs

### PR DESCRIPTION
**Pull request by michsa**
Updates storybook to 5.2, and reorganizes our stories to work better with the new storybook autodocs. Generally this means each component is now its own storybook entity thing, even if it only contains a single story.

Before:
Screen Shot 2019-12-11 at 11 30 23 AM

After:
Screen Shot 2019-12-11 at 11 31 03 AM

I also converted a couple of files to the new component story format: Awaiter, Box, Button, Checkbox, ErrorList, Modal, and Popover. The eventual goal is a consistent format across all our stories - which will probably be MDX, but I figured I'd leave these files in CSF for demonstration.

For the stories which were left in the old-school format, we no longer need `addWithJSX` (in fact, it's broken now), so I replaced it with just `add` throughout.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR has lint errors. Please fix these and commit them.
      
<details>
  <summary>config.js</summary>
  <p><a href="https://github.com/smartprocure/contexture-react/blob/v2-storydocs/s" target="_blank">s</a></p>
  
  <strong>'addParameters' is defined but never used.</strong>
  <code>Line 1: import { configure, addParameters } from '@storybook/react'</code>

  <strong>'DocsPage' is defined but never used.</strong>
  <code>Line 2: import { DocsPage } from '@storybook/addon-docs/blocks'</code>

  <strong>'PropsTable' is defined but never used.</strong>
  <code>Line 3: import { PropsTable } from '@storybook/components'</code>

  <strong>'F' is defined but never used.</strong>
  <code>Line 4: import F from 'futil-js'</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+6555 -2357)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Could not run `duti:fix` command successfully
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 419be51efce1e34c05b84387b3aa439625dc0dbc
</p>

File: src/stories/devNotes.stories.js [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-      text: require('../../CHANGELOG.md'),
-      inline: true,
-      source: false,
-      header: false,
-    })(() => null)
-  )
-  .add(
-    'Theme API',
-    withInfo({
-      text: require('../../docs/theme.md'),
```
Comment (daedalus28) [inline src/stories/devNotes.stories.js]: this should live somewhere in the storybook

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+6658 -2440)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 5af1121bb8556f91369d0c67bdc2e22112975562


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9dc1aba9a49d8c17b52ecb4b25fb161000ddde32
</p>

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+6703 -2492)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 63ad8e9f204975783de8038c70c9eb30f139dfc9
</p>

**Pull request merged**